bwe/RateController.cpp

```diff
@@ -226,15 +224,22 @@ bool isProbe(const T& backlog, double bandwidthKbps, uint32_t trainEnd, uint32_t
     for (uint32_t i = srIndex - 1;; --i)
     {
         const auto& item = backlog[i];
-        queue.onPacketSent(item.transmissionTime, item.size);
-        if (queue.size() <= item.size)
+
+        // It's important to drain and check before add packet. Otherwise we can't be sure we went
+        // lower than initial size because adding a new packet in the most of cases will increase the queue
+        // above initial
+        queue.drain(item.transmissionTime);
```

Comment:
drain first! otherwise we just need the packets added to have a bigger size that the SR